optaplanner-integration/optaplanner-spring-boot-autoconfigure/src/main/java/org/optaplanner/spring/boot/autoconfigure/OptaPlannerAutoConfiguration.java

```diff
@@ -189,11 +185,46 @@ private void applySolverProperties(SolverConfig solverConfig) {
         }
         if (entityClassSet.isEmpty()) {
             throw new IllegalStateException("No classes (" + entityClassSet
-                    + ") found with a @" + PlanningEntity.class.getSimpleName() + " annotation.");
+                    + ") found with a @" + PlanningEntity.class.getSimpleName() + " annotation.\n"
+                    + "Maybe your @" + PlanningEntity.class.getSimpleName() + " annotated class(es) "
+                    + " are not in a subpackage of your @" + SpringBootApplication.class.getSimpleName()
+                    + " annotated class's package.\n"
+                    + "Maybe move your planning entity classes to your application class's (sub)package"
+                    + " (or use @" + EntityScan.class.getSimpleName() + ").");
         }
         return new ArrayList<>(entityClassSet);
     }
 
+    private void applyScoreDirectorFactoryProperties(SolverConfig solverConfig) {
+        if (solverConfig.getScoreDirectorFactoryConfig() == null) {
+            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = new ScoreDirectorFactoryConfig();
+            scoreDirectorFactoryConfig.setEasyScoreCalculatorClass(findImplementingClass(EasyScoreCalculator.class));
+            scoreDirectorFactoryConfig.setConstraintProviderClass(findImplementingClass(ConstraintProvider.class));
+            scoreDirectorFactoryConfig
+                    .setIncrementalScoreCalculatorClass(findImplementingClass(IncrementalScoreCalculator.class));
+            if (beanClassLoader.getResource(SolverProperties.DEFAULT_SCORE_DRL_URL) != null) {
+                scoreDirectorFactoryConfig.setScoreDrlList(Collections.singletonList(
+                        SolverProperties.DEFAULT_SCORE_DRL_URL));
+            }
+            if (scoreDirectorFactoryConfig.getEasyScoreCalculatorClass() == null
+                    && scoreDirectorFactoryConfig.getEasyScoreCalculatorClass() == null
```

Comment:
The same condition repeated. Spotbugs maven plugin failed the build due to this:

09:58:13 [ERROR] Repeated conditional test in org.optaplanner.spring.boot.autoconfigure.OptaPlannerAutoConfiguration.applyScoreDirectorFactoryProperties(SolverConfig) [org.optaplanner.spring.boot.autoconfigure.OptaPlannerAutoConfiguration, org.optaplanner.spring.boot.autoconfigure.OptaPlannerAutoConfiguration] At OptaPlannerAutoConfiguration.java:[line 209]At OptaPlannerAutoConfiguration.java:[lines 209-210] RpC_REPEATED_CONDITIONAL_TEST

Reply:
Good catch, fixing